SDKTest/NIDPerformanceTests.swift

```diff
@@ -33,70 +32,65 @@ final class NIDPerformanceTests: XCTestCase {
         clearOutDataStore()
     }
 
-    // Commenting the following tests out as they seem to be a little unstable
-    //  additionally they interfere with the other tests (filling up the data store)
-    //  once we can reliably run these tests we can uncomment them
-//
-//    func testBufferFull() throws {
-//        for _ in 1...3000 {
-//            let expectedValue = "myTestUserID"
-//            _ = NeuroID.setGenericUserID(
-//                userId: expectedValue,
-//                type: .registeredUserID
-//            ) { res in
-//                res
-//            }
-//        }
-//        print("NID Event Size: \(DataStore.events.count)")
-//        assert(DataStore.events.count <= 2010)
-//        assert(DataStore.events.count >= 1)
-//        assert(DataStore.events.last!.type == NIDEventName.bufferFull.rawValue)
-//    }
-//
-//    func testQueuedEvents() throws {
-//        _ = NeuroID.stop()
-//        for _ in 1...2100 {
-//            let expectedValue = "myTestUserID"
-//            _ = NeuroID.setGenericUserID(
-//                userId: expectedValue,
-//                type: .registeredUserID
-//            ) { res in
-//                res
-//            }
-//        }
-//        print("NID Queue Size: \(DataStore.queuedEvents.count)")
-//        assert(DataStore.queuedEvents.count <= 2010)
-//        assert(DataStore.queuedEvents.count >= 1)
-//        assert(DataStore.queuedEvents.last?.type == NIDEventName.bufferFull.rawValue)
-//    }
-//
-//    func testLowMemory() throws {
-//        // Setup a view and invoke observeAppEvents to get listeners attached for the test
-//        let uiView = UITextView()
-//
-//        let screenNameValue = "testScreen"
-//
-//        let tracker = NeuroIDTracker(screen: "Temp", controller: uiView.inputViewController)
-//
-//        let guidValue = "\(Constants.attrGuidKey.rawValue)"
-//
-//        tracker.observeAppEvents()
-//
-//        NeuroIDTracker.registerSingleView(v: uiView, screenName: screenNameValue, guid: guidValue)
-//
-//        // Manually trigger low memory event
-//        NotificationCenter.default.post(name: UIApplication.didReceiveMemoryWarningNotification, object: nil)
-//
-//        for _ in 1...100 {
-//            let expectedValue = "myTestUserID"
-//            _ = NeuroID.setGenericUserID(
-//                userId: expectedValue,
-//                type: .registeredUserID
-//            ) { res in
-//                res
-//            }
-//        }
-//        print("NID Event Size low Memory: \(DataStore.events.count)")
-//        assert(DataStore.events.count == 0)
-//    }
+    /**
+     Perforamnce tests are in one test to ensure test is not async with others and to prevent hanging*/
+    func all_performance_tests() throws {
```

Comment:
Will this run the test alone either before or after the others? If it still runs concurrently with others then we can have the same issue because DataStore is a static instance everything is accessing. Consider leaving this out until we can verify for sure?

Reply:
Part of issue was just the loop that was filling up events and those running in parallel.

Reply:
Right, but if this test can still be run in parallel then we are still going to encounter a potential race condition where this one test is running while another one is, we have lessened the surface area by putting it into one but this test still access and fills the same data store

Reply:
No way to run this test first/last/ relibably from my research. We can also just not use this test suite.

Reply:
I vote that for now we leave out this test suite, when we refactor to use dependency injection and a class instance of DataStore instead of static this tests will be valid and very useful